### PR TITLE
two bugs in grouping and training were fixed

### PR DESCRIPTION
**Pull request by Katy-katy**
@lensonp I tried to train the models using flow_reactor_modeling_dataset and found 2 bugs:
1. classifier.py: for example, we have crystalline 178 False and 1 True.  In the lines 100-106 we remove the labels with count < 5. Now we have only "False". So, after removing labels we need to check one more time if we have at least 2 labels. 

2. train.py: it is possible that our training data does not includes all "flags".  So, we should train the model only for flags that exist in training data.

### CODE REVIEW

**Comment by lensonp** [comment]
@Katy-katy - Ok. I'll look at this in a second. However, I think that the one crystalline sample was flagged with good_fit==False. Shouldn't it have been filtered out of the training set before training?

**Comment by lensonp** [comment]
@Katy-katy - This is now fixed, I think. I also had to include the parameter label fix from #191 in order to test the training. We will have to rebase or resolve conflicts for #191.
